### PR TITLE
Refactor hi-pass viz code

### PR DESCRIPTION
**Pull request by johannesring**
In this pull request, I have cleaned up the hi-pass viz code, added some documentation, and added a script `fsipy-create-hi-pass-viz`.

### CODE REVIEW

File: src/fsipy/automatedPostprocessing/postprocessing_h5py/postprocessing_common_h5py.py [hunk 1]
```diff
@@ -384,3 +343,305 @@ def create_transformed_matrix(input_path: Union[str, Path], output_folder: Union
...
+    # Pad the RMS array to match the length of the original signal
+    RMS_padded = np.zeros(len_signal_squared)
+    for i in range(len_signal_squared):
+        if pad_length <= i < len_RMS + pad_length:
+            RMS_padded[i] = RMS[i - pad_length]
+
+    return RMS_padded
+
+
+def get_eig(T: np.ndarray) -> float:
```
Comment (keiyamamo) [inline src/fsipy/automatedPostprocessing/postprocessing_h5py/postprocessing_common_h5py.py:579]: This function exists in turtle, so you can just import.

https://github.com/KVSlab/turtleFSI/blob/286fab226acb6fa35cf83cef95ef5cdd454026b7/turtleFSI/modules/common.py#L148

**Review by keiyamamo — COMMENTED** [review]
Hi @johannesring 

Thanks for the PR! I just have one quick comment and I will test them locally and get back to you!

**Comment by keiyamamo** [comment]
Hi @johannesring,

I tested the high-pass-viz functionality and it works very well with displacement and velocity. For the pressure I had to specify `save-deg=1` since pressure is (almost) always P1 and `save-deg` function does not affect pressure. It might be nice if we could assume `save-deg=1` when `-q p`, but it is a minor comment.

I also tested with my new stress/strain implementation and it needs some updates due to the changes I made with `write_checkpoint` as well as some naming of the xdmf files. I’m investigating if I can fix easily but it seems to require some time since the way `h5` file and `xdmf` file are generated is different from those generated with `write`. 

For example, I changed `output_file_lists` function as follows for it to work. 


```
def output_file_lists(xdmf_file):
    """
    If the simulation has been restarted, the output is stored in multiple files and may not have even temporal spacing
    This loop determines the file names from the xdmf output file

    Args:
        xdmf_file (Path): Path to xdmf file

    Returns:
        h5file_name_list (list): List of names of h5 files associated with each timestep
        timevalue_list (list): List of time values in xdmf file
        index_list (list): List of indices of each timestp in the corresponding h5 file
    """

    file1 = open(xdmf_file, 'r')
    Lines = file1.readlines()
    h5file_name_list = []
    timevalue_list = []
    index_list = []

    # This loop goes through the xdmf output file and gets the time value (timevalue_list), associated
    # with .h5 file (h5file_name_list) and index of each timestep in the corresponding h5 file (index_list)
    for line in Lines:

        if "FiniteElementFunction" in line:
            checkpoint_data = True
            print("xdmf file was generated with write_checkpoint")
            break
        else:
            checkpoint_data = False

    if checkpoint_data:
        for line in Lines:
            if '<Time Value' in line:
                time_pattern = '<Time Value="(.+?)"'
                time_str = re.findall(time_pattern, line)
                time = float(time_str[0])
                timevalue_list.append(time)

            elif 'vector' in line:
                h5_pattern = r'"HDF">(.*?):'
                h5_str = re.findall(h5_pattern, line)
                h5file_name_list.append(h5_str[0])

                index_pattern = r'_([0-9]+)\/vector'
                index_str = re.findall(index_pattern, line)
                index = int(index_str[0])
                index_list.append(index)
    else:
        for line in Lines:
            if '<Time Value' in line:
                time_pattern = '<Time Value="(.+?)"'
                time_str = re.findall(time_pattern, line)
                time = float(time_str[0])
                timevalue_list.append(time)

            elif 'VisualisationVector' in line:
                h5_pattern = '"HDF">(.+?):/'
                h5_str = re.findall(h5_pattern, line)
                h5file_name_list.append(h5_str[0])

                index_pattern = "VisualisationVector/(.+?)</DataItem>"
                index_str = re.findall(index_pattern, line)
                index = int(index_str[0])
                index_list.append(index)

    return h5file_name_list, timevalue_list, index_list
```

and I’m sure we need to update `create_transformed_matrix` function as well. 

Do you think it would be possible for you to update high-pass-viz functionality so that it can work with new stress/strain implementation? If there are other things you would like to work on, we can merge this PR and I can work on updating high-pass-viz functionality. Please let me know what you think. Also, I can share `xdmf` file generated with new stress/strain code if you need them for test, so let me know about it as well. 

Best,
Kei

**Comment by keiyamamo** [comment]
Hi @johannesring 

> I assume `WSS_ts.xdmf` is now `WSS.xdmf` and `MaxPrincipalStrain.xdmf` is the same as before, but what about `InfinitesimalStrain.xdmf`?

Yes, `WSS_ts.xdmf` is now renamed as `WSS.xdmf`. This is to be consistent with `VaMPy` naming. For `InfinitesimalStrain.xdmf`, it is changed to `Green-Lagrange-strain.xdmf`.

> I will try to get it to work with your new implementation, but I don't want to spend too much time on it since I also want to finish the extract branch feature (for AVF's) in the preprocessing script before I go on vacation.

Thanks for the answer! I agree with you that it would be nice to finish extract branch feature before you go on vacation. I have no problem working on changing high-pass-viz with new xdmf files because I’m mostly done with changing FEniCS code. So please feel free to focus on the branch feature and let me know if you want to work on high-pass-viz from now on. 

> I think I have generated the new xdmf files already (as listed above), but let me know if there are more files.

You have all you need :)

**Comment by johannesring** [comment]
> Thanks for the answer! I agree with you that it would be nice to finish extract branch feature before you go on vacation. I have no problem working on changing high-pass-viz with new xdmf files because I’m mostly done with changing FEniCS code. So please feel free to focus on the branch feature and let me know if you want to work on high-pass-viz from now on.

I started working on this, but I will leave the rest up to you @keiyamamo. I will instead focus on the extract branch feature.

**Comment by keiyamamo** [comment]
@johannesring  Thanks for letting me know! I will take over the branch and will work on the rest :)

**Comment by keiyamamo** [comment]
@johannesring I’ve finished updating the high pass viz script. Could you please check it when you have time?

**Comment by johannesring** [comment]
This looks good to me. I have also tested the script with different command line options and it works well.

**Comment by keiyamamo** [comment]
@johannesring  Thank you! I just found one thing to update, so will add a commit, and then merge after that.

**Pull request merged**